### PR TITLE
Add support for mode, locale, userReferralCode options

### PR DESCRIPTION
**Pull request by ndhoule**
Making this change on behalf of @tsholmes, who doesn't yet have GitHub permissions on this org.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -54,6 +54,8 @@ SaaSquatch.prototype.identify = function(identify) {
   var paymentProviderId = identify.proxy('traits.paymentProviderId');
   var accountStatus = identify.proxy('traits.accountStatus');
   var referralCode = identify.proxy('traits.referralCode');
+  var userReferralCode = identify.proxy('traits.userReferralCode');
```
Comment (ndhoule) [inline lib/index.js]: cc @sperand-io on this comment:

Doesn't this encourage users to pass a saasquatch-specific option in user traits? Should we have users pass this via `context.integrations`, instead?

(I realize this is following example, but I'd like to understand the ideal)

Comment (f2prateek) [inline lib/index.js]: Should we send all the traits along as is to the tool, while mapping any special ones that are match our and their spec?

Comment (sperand-io) [inline lib/index.js]: +1 +1 (both comments) — let's send through as many traits as possible (if they accept arbitrary user attributes) and use `context.SaaSquatch.$` for the `userReferralCode` field. @tsholmes happy to go over why in person when I get back from SEA!

**Comment by f2prateek** [comment]
Nice this looks great! LGTM

**Comment by sperand-io** [comment]
LGTM!

**Pull request merged**